drivers/sht1x/sht1x.c

```diff
@@ -0,0 +1,665 @@
+int16_t sht1x_humidity(const sht1x_dev_t *dev, uint16_t raw, int16_t temp)
+{
+    if (!dev)
+        return -1;
+
+    static const int32_t c1 = -20468;
+    static const int32_t t1 = 1;
+    int32_t c2, c3, c4, t2;
+    if (dev->conf & SHT1X_CONF_LOW_RESOLUTION){
+        c2 = 5872;
+        c3 = 494801;
+        c4 = 1000000;
+        t2 = 781;
+    }
+    else {
+        c2 = 367;
+        c3 = 791684;
+        c4 = 100000;
+        t2 = 12500;
+    }
+
+    /*
+     * Calculate linear humidity, but slightly different. Original formula:
+     *
+     * hum_lin = c1 + c2 * raw + c3 * (raw * raw)
+     *
+     * But we use:
+     *
+     * hum_lin = c1 + c2 * raw - (c4 * raw / c3') * (c4 * raw / c3')
+     *
+     * where: c3' = 1 / (sqrt(-c3) / c4)
+     *
+     * (This better fits for integer calculation)
+     */
+
+    int32_t res = ((int32_t)raw * c4) / c3;
+    res = c1 + c2 * (int32_t)raw - (res * res);
+
+    /*
+     * Perform temperature compensation, again slightly different.
+     * Original formula:
+     *
+     * hum_true = (temp - 25) * (t1 + t2 * raw) + hum_lin
+     *
+     * But we use:
+     *
+     * hum_true = (temp - 25) * t1 + (temp - 25) * raw / t2') + hum_lin
+     *
+     * where t2' = 1/t2
+     */
+    int32_t temp_diff = temp - 2500;
+    res = temp_diff * t1 + (temp_diff * (int32_t)raw * 100) / t2 + res;
+    return (int16_t)(res/100);
+}
+
+
+/*---------------------------------------------------------------------------*/
+int sht1x_read(const sht1x_dev_t *dev, int16_t *temp, int16_t *rel_hum)
+{
+    uint16_t temp_raw;
+    int16_t t;
+    uint16_t hum_raw;
+    int retval;
+
+    if (
+        !dev ||
+        (dev->vdd >= sizeof(sht1x_d1)/sizeof(sht1x_d1[0])) ||
+        (!temp && !rel_hum)
+    )
+        return -EINVAL;
+
+    retval = measure(dev, &temp_raw, SHT1X_MEASURE_TEMP);
+    if (retval != 0){
+        connection_reset(dev);
+        return retval;
+    }
+
+    t = sht1x_temperature(dev, temp_raw);
+
+    if (temp != NULL)
+        *temp = t;
+
+    if (rel_hum != NULL){
+        retval = measure(dev, &hum_raw, SHT1X_MEASURE_HUM);
+        if (retval != 0){
+            connection_reset(dev);
+            return retval;
+        }
+
+        *rel_hum = sht1x_humidity(dev, hum_raw, t);
+    }
+
+    return 0;
+}
+
+/*---------------------------------------------------------------------------*/
+int sht1x_configure(sht1x_dev_t *dev, sht1x_conf_t conf)
+{
+    if (!dev)
+        return -EINVAL;
+
+    /* Apply config that is not stored on the sensor */
+    dev->conf &= SHT1X_CONF_MASK;
+    dev->conf |= conf & (~(SHT1X_CONF_MASK));
+
+    /* Send new status byte to sensor, if on-device config was changed */
+    if ((conf & SHT1X_CONF_MASK) != (dev->conf & SHT1X_CONF_MASK)) {
+        int retval = transmission_start(dev);
+        if (retval != 0)
+            return retval;
+
+        switch (write_byte(dev, SHT1X_STATUS_REG_W)){
```

Comment:
Maybe this file could be uncrustified ? (a space is missing before `{` here in a lot of other places).